### PR TITLE
JBDS-4438 Detect free space available on disk for Windows

### PR DESCRIPTION
**Pull request by sudhirverma**

### CODE REVIEW

File: browser/services/platform.js [hunk 1]
```diff
@@ -133,6 +133,35 @@ class Platform {
     });
   }
 
+  static getAvailableDisk(locations) {
```
Comment (dgolovin) [inline browser/services/platform.js]:
``` Javascript
  static getFreeDiskSpace(location) {
    return Platform.identify({
      win32: ()=> {
        let disk = path.parse(location).root.charAt(0);
        return pify(child_process.exec)(`powershell -command --% (Get-WMIObject Win32_Logicaldisk -filter \"deviceid=\`'${disk}:\`'\").freespace`).then((output)=>{
          return new Promise.resolve(Number.parseInt(output));
        });
      },
      darwin: ()=> {
          //different story
      }
    });
  }
```

Comment (dgolovin) [inline browser/services/platform.js]: For macOS it seems you can use
``` shell
df -k /path/to/target/install folder
```
then you have to parse output.

``` shell
Filesystem   1024-blocks      Used Available Capacity  Mounted on
/dev/disk0s2   244781464 173055384  71470080    71%    /
```

Comment (dgolovin) [inline browser/services/platform.js]:
``` javascript
var child_process = require('child_process');
require('pify')(child_process.exec)(
'powershell.exe -command "& {(Get-WMIObject Win32_Logicaldisk -filter \\"deviceid=\'c:\'\\").freespace }"'
).then((output)=>{
  console.log(output.trim());
});
```
save snipped above to test.js file and then run it

``` 
node test.js
```

**Comment by sudhirverma** [comment]
test this

File: browser/services/platform.js [hunk 2]
```diff
@@ -133,6 +133,23 @@ class Platform {
...
   }
 
+  static getFreeDiskSpace(location) {
+    return Platform.identify({
+      win32: ()=> {
+        let disk = path.parse(location).root.charAt(0);
+        return pify(child_process.exec)(`powershell -command "& {(Get-WMIObject Win32_Logicaldisk -filter \"deviceid=\`'${disk}:\`'\") }"`).then((stdout) => {
+          return Promise.resolve(Number.parseInt(stdout));
+        }).catch(()=>{
+          return Promise.resolve();
```
Comment (dgolovin) [inline browser/services/platform.js]: The error should be logged at least.

File: test/unit/services/platform-test.js [hunk 3]
```diff
@@ -343,6 +343,23 @@ describe('Platform', function() {
...
+      });
+
+      it('should able to return free disk space', function() {
+        sandbox.stub(child_process, 'exec').yields(undefined, '248005160960');
+        let location = 'C:\\DevelopmentSuite';
+        return Platform.getFreeDiskSpace(location).then((result) => {
+          expect(result).to.be.equal(248005160960);
+        });
+      });
+    });
```
Comment (dgolovin) [inline test/unit/services/platform-test.js]: There should be test for execution path with error.

Comment (dgolovin) [inline test/unit/services/platform-test.js]: It also should test command actually parsing location passed as a parameter and puts correct disk name in to executed command.

**Comment by dgolovin** [comment]
test this

File: test/unit/services/platform-test.js [hunk 4]
```diff
@@ -358,10 +357,16 @@ describe('Platform', function() {
           expect(result).to.be.equal(248005160960);
         });
       });
+
+      it('should return NaN if drive not found', function() {
+        sandbox.stub(child_process, 'exec').yields(undefined, "");
+        let location = 'd:\\DevelopmentSuite';
+        return Platform.getFreeDiskSpace(location).then((result) => {
+          expect(result).to.be.NaN;
+      });
```
Comment (jrichter1) [inline test/unit/services/platform-test.js]: I think you forgot to close one parentheses combo after the then callback. There's a syntax error now.

**Comment by sudhirverma** [comment]
test this

**Pull request closed without merging**